Impeller uses deprecated AChoreographer_postFrameCallback

Attempting to build the engine against android_apl_level 35, the minimum for RISC-V, results in

```
../../flutter/impeller/toolkit/android/proc_table.h:124:3: warning: 'AChoreographer_postFrameCallback' is deprecated: first deprecated in Android 29 - Use AChoreographer_postFrameCallback64 instead [-Wdeprecated-declarations]
  124 |   FOR_EACH_ANDROID_PROC(DEFINE_PROC);
      |   ^
../../flutter/impeller/toolkit/android/proc_table.h:42:10: note: expanded from macro 'FOR_EACH_ANDROID_PROC'
   42 |   INVOKE(AChoreographer_postFrameCallback, 24)                   \
      |          ^
../../flutter/third_party/android_tools/ndk/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/android/choreographer.h:157:6: note: 'AChoreographer_postFrameCallback' has been explicitly marked deprecated here
  157 | void AChoreographer_postFrameCallback(AChoreographer* choreographer,
      |      ^
1 warning generated.
```

Unfortunately, the engine treats deprecation notices as errors, so this is fatal.

--- a/engine/src/flutter/impeller/toolkit/android/proc_table.h
+++ b/engine/src/flutter/impeller/toolkit/android/proc_table.h
@@ -39,10 +39,10 @@ ASurfaceTransaction* ASurfaceTransaction_fromJava(JNIEnv* env,
 ///
 #define FOR_EACH_ANDROID_PROC(INVOKE)                            \
   INVOKE(AChoreographer_getInstance, 24)                         \
-  _Pragma("clang diagnostic push")                               \
-  _Pragma("clang diagnostic ignored \"-Wdeprecated-declarations\"") \
+  _Pragma("GCC diagnostic push")                               \
+  _Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"") \
   INVOKE(AChoreographer_postFrameCallback, 24)                   \
-  _Pragma("clang diagnostic pop")                                \
+  _Pragma("GCC diagnostic pop")                                \
   INVOKE(AChoreographer_postFrameCallback64, 29)                 \
   INVOKE(AHardwareBuffer_acquire, 26)                            \
   INVOKE(AHardwareBuffer_allocate, 26)                           \

--- a/engine/src/flutter/impeller/toolkit/android/proc_table.h
+++ b/engine/src/flutter/impeller/toolkit/android/proc_table.h
@@ -39,7 +39,10 @@ ASurfaceTransaction* ASurfaceTransaction_fromJava(JNIEnv* env,
 ///
 #define FOR_EACH_ANDROID_PROC(INVOKE)                            \
   INVOKE(AChoreographer_getInstance, 24)                         \
+  _Pragma("clang diagnostic push")                               \
+  _Pragma("clang diagnostic ignored \"-Wdeprecated-declarations\"") \
   INVOKE(AChoreographer_postFrameCallback, 24)                   \
+  _Pragma("clang diagnostic pop")                                \
   INVOKE(AChoreographer_postFrameCallback64, 29)                 \
   INVOKE(AHardwareBuffer_acquire, 26)                            \
   INVOKE(AHardwareBuffer_allocate, 26)                           \

--- a/engine/src/flutter/impeller/toolkit/android/proc_table.h
+++ b/engine/src/flutter/impeller/toolkit/android/proc_table.h
@@ -37,39 +37,41 @@ ASurfaceTransaction* ASurfaceTransaction_fromJava(JNIEnv* env,
 ///             directly. Instead, rely on the handle wrappers (`Choreographer`,
 ///             `HardwareBuffer`, etc..).
 ///
-#define FOR_EACH_ANDROID_PROC(INVOKE)                            \
-  INVOKE(AChoreographer_getInstance, 24)                         \
-  _Pragma("GCC diagnostic push")                               \
-  _Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"") \
-  INVOKE(AChoreographer_postFrameCallback, 24)                   \
-  _Pragma("GCC diagnostic pop")                                \
-  INVOKE(AChoreographer_postFrameCallback64, 29)                 \
-  INVOKE(AHardwareBuffer_acquire, 26)                            \
-  INVOKE(AHardwareBuffer_allocate, 26)                           \
-  INVOKE(AHardwareBuffer_describe, 26)                           \
-  INVOKE(AHardwareBuffer_fromHardwareBuffer, 26)                 \
-  INVOKE(AHardwareBuffer_getId, 31)                              \
-  INVOKE(AHardwareBuffer_isSupported, 29)                        \
-  INVOKE(AHardwareBuffer_lock, 26)                               \
-  INVOKE(AHardwareBuffer_release, 26)                            \
-  INVOKE(AHardwareBuffer_unlock, 26)                             \
-  INVOKE(ANativeWindow_acquire, 0)                               \
-  INVOKE(ANativeWindow_getHeight, 0)                             \
-  INVOKE(ANativeWindow_getWidth, 0)                              \
-  INVOKE(ANativeWindow_release, 0)                               \
-  INVOKE(ASurfaceControl_createFromWindow, 29)                   \
-  INVOKE(ASurfaceControl_release, 29)                            \
-  INVOKE(ASurfaceTransaction_apply, 29)                          \
-  INVOKE(ASurfaceTransaction_create, 29)                         \
-  INVOKE(ASurfaceTransaction_delete, 29)                         \
-  INVOKE(ASurfaceTransaction_reparent, 29)                       \
-  INVOKE(ASurfaceTransaction_setBuffer, 29)                      \
-  INVOKE(ASurfaceTransaction_setColor, 29)                       \
-  INVOKE(ASurfaceTransaction_setOnComplete, 29)                  \
-  INVOKE(ASurfaceTransactionStats_getPreviousReleaseFenceFd, 29) \
-  INVOKE(ASurfaceTransaction_fromJava, 34)                       \
-  INVOKE(ATrace_isEnabled, 23)                                   \
+// clang-format off
+#define FOR_EACH_ANDROID_PROC(INVOKE)                              \
+  INVOKE(AChoreographer_getInstance, 24)                           \
+  _Pragma("GCC diagnostic push")                                   \
+  _Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"")  \
+  INVOKE(AChoreographer_postFrameCallback, 24)                     \
+  _Pragma("GCC diagnostic pop")                                    \
+  INVOKE(AChoreographer_postFrameCallback64, 29)                   \
+  INVOKE(AHardwareBuffer_acquire, 26)                              \
+  INVOKE(AHardwareBuffer_allocate, 26)                             \
+  INVOKE(AHardwareBuffer_describe, 26)                             \
+  INVOKE(AHardwareBuffer_fromHardwareBuffer, 26)                   \
+  INVOKE(AHardwareBuffer_getId, 31)                                \
+  INVOKE(AHardwareBuffer_isSupported, 29)                          \
+  INVOKE(AHardwareBuffer_lock, 26)                                 \
+  INVOKE(AHardwareBuffer_release, 26)                              \
+  INVOKE(AHardwareBuffer_unlock, 26)                               \
+  INVOKE(ANativeWindow_acquire, 0)                                 \
+  INVOKE(ANativeWindow_getHeight, 0)                               \
+  INVOKE(ANativeWindow_getWidth, 0)                                \
+  INVOKE(ANativeWindow_release, 0)                                 \
+  INVOKE(ASurfaceControl_createFromWindow, 29)                     \
+  INVOKE(ASurfaceControl_release, 29)                              \
+  INVOKE(ASurfaceTransaction_apply, 29)                            \
+  INVOKE(ASurfaceTransaction_create, 29)                           \
+  INVOKE(ASurfaceTransaction_delete, 29)                           \
+  INVOKE(ASurfaceTransaction_reparent, 29)                         \
+  INVOKE(ASurfaceTransaction_setBuffer, 29)                        \
+  INVOKE(ASurfaceTransaction_setColor, 29)                         \
+  INVOKE(ASurfaceTransaction_setOnComplete, 29)                    \
+  INVOKE(ASurfaceTransactionStats_getPreviousReleaseFenceFd, 29)   \
+  INVOKE(ASurfaceTransaction_fromJava, 34)                         \
+  INVOKE(ATrace_isEnabled, 23)                                     \
   INVOKE(eglGetNativeClientBufferANDROID, 0)
+// clang-format on
 
 template <class T>
 struct AndroidProc {

